memory/src/main/java/org/opensearch/ml/memory/action/conversation/CreateConversationRequest.java

```diff
@@ -158,9 +176,11 @@ public static CreateConversationRequest fromRestRequest(RestRequest restRequest)
                 }
             }
             if (body.get(ActionConstants.REQUEST_CONVERSATION_NAME_FIELD) != null) {
-                return new CreateConversationRequest(name, applicationType, additionalInfo);
+                return new CreateConversationRequest(name, applicationType, additionalInfo, true);
             } else {
-                return new CreateConversationRequest();
+                CreateConversationRequest request = new CreateConversationRequest();
+                request.fromRest = true;
```

Comment:
nit: instead setting fromRest like this we can have a constructor to pass along fromRest